contracts/router/interfaces/IRouterV2.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {BridgeToken, DestRequest, SwapQuery} from "../libs/Structs.sol";
+
+interface IRouterV2 {
+    /// @notice Initiate a bridge transaction with an optional swap on both origin and destination chains.
+    /// @dev Note that method is payable.
+    /// If token is ETH_ADDRESS, this method should be invoked with `msg.value = amountIn`.
+    /// If token is ERC20, the tokens will be pulled from msg.sender (use `msg.value = 0`).
+    /// Make sure to approve this contract for spending `token` beforehand.
+    /// originQuery.tokenOut should never be ETH_ADDRESS, bridge only works with ERC20 tokens.
+    ///
+    /// `token` is always a token user is sending. In case token requires a wrapper token to be bridge,
+    /// use underlying address for `token` instead of the wrapper one.
+    ///
+    /// `originQuery` contains instructions for the swap on origin chain. As above, originQuery.tokenOut
+    /// should always use the underlying address. In other words, the concept of wrapper token is fully
+    /// abstracted away from the end user.
+    ///
+    /// `originQuery` is supposed to be fetched using SynapseRouter.getOriginAmountOut().
+    /// Alternatively one could use an external adapter for more complex swaps on the origin chain.
+    ///
+    /// `destQuery` is supposed to be fetched using SynapseRouter.getDestinationAmountOut().
+    /// Complex swaps on destination chain are not supported for the time being.
+    /// Check contract description above for more details.
+    ///
+    /// @param to            Address to receive tokens on destination chain
+    /// @param chainId       Destination chain id
+    /// @param moduleId      Bridge module id to delegate bridge call
+    /// @param token         Initial token for the bridge transaction to be pulled from the user
+    /// @param amount        Amount of the initial tokens for the bridge transaction
+    /// @param originQuery   Origin swap query. Empty struct indicates no swap is required
+    /// @param destQuery     Destination swap query. Empty struct indicates no swap is required
+    function bridgeViaSynapse(
+        address to,
+        uint256 chainId,
+        bytes32 moduleId,
+        address token,
+        uint256 amount,
+        SwapQuery memory originQuery,
+        SwapQuery memory destQuery
+    ) external payable;
+
```

Comment:
Let's add the swap function as well:

https://github.com/synapsecns/synapse-contracts/blob/8ee29a886b3cfce25cae786075d474f75baaec96/contracts/bridge/router/SynapseRouter.sol#L216-L235